core/aws-lsp-core/src/util/workspaceUtils.ts

```
@@ -99,12 +99,16 @@ export function formatListing(entry: Dirent): string {
 }
 
 export function getEntryPath(entry: Dirent) {
+    if (!entry.parentPath) {
```

doesn't typescript satisfy this automatically since `parentpath` is a non-optional field in Dirent? 

The only place where the parentpath was not being set before was in the testing stubs that had custom behaviour and that were doing lots of type casting. Throwing and error here doesn't look really right to me, now with the parentpath changes introduced in the PR do we need to throw this error?

---

I will remove the throw and error check in getEntryPath as I was seeing errors locally where parentpath was null with tests. I wasn't sure if this only impacted tests but as long as it does not impact actual logic, I am fine with removing this